include/dpp/webhook.h

```diff
@@ -56,6 +56,24 @@ class DPP_EXPORT webhook : public managed {
 	 */
 	webhook();
 
+	/**
+     * @brief Construct a new webhook object using the Webhook URL provided by Discord
+    */
+	webhook(std::string webhook_url)
+	{
+		token = webhook_url.substr(webhook_url.find_last_of("/") + 1); //find token
+		id = std::stoull(webhook_url.substr(sizeof("https://discord.com/api/webhooks/") - 1, 18)); //18 is the amount of characters a snowflake possesses
```

Comment:
A snowflake doesn't necessarily be 18 characters. There are 17 digit ids, and soon there will be 19 digit ids.

Reply:
Noted that, thanks for pointing it out!

Reply:
also bear in mind std::stoull can throw... if you don't want to bubble that exception to the user you should catch it, and probably re throw it as `dpp::logic_exception`

Reply:
also ctor body belongs in the cpp or you can end up with it being defined more than once causing a compile error